tests/get-coverage.yml

```diff
@@ -0,0 +1,73 @@
+# SPDX-License-Identifier: BSD-3-Clause
+---
+- name: Prepare for coverage extraction
+  hosts: all
+  tasks:
+    - name: set facts
+      set_fact:
+        coverage_module: network_connections
+        coverage: /root/.local/bin/coverage
+    - name: set more facts
+      set_fact:
+        coverage_file: ansible-coverage-{{ coverage_module }}-{{ test_playbook|replace('.yml', '') }}
+
+    - name: set more facts
+      set_fact:
+        destdir: "remote_coverage/{{ test_playbook }}"
+      when: destdir is undefined
+
+    - name: debug info
+      debug:
+        msg: Getting coverage for '{{ coverage_module }}' with '{{ test_playbook }}'
+
+    # combine data in case old data is left there
+    - command: "{{ coverage }} combine"
+      environment:
+        COVERAGE_FILE: "{{ coverage_file }}"
+      ignore_errors: yes
+
+    - name: remove old data
+      file:
+        state: absent
+        path: "{{ item }}"
+      loop:
+        - .coverage
+
+    - name: remove test data
+      shell: rm -rf /root/.ansible/ /root/.coverage.* /root/htmlcov /tmp/ansible_*
+
+    - name: copy coveragerc
+      copy:
+        content: "[run]\ndisable_warnings = no-data-collected\n"
+        dest: .coveragerc
+
+    - name: install latest pip
+      pip:
+        name: coverage
+        extra_args: --user --upgrade
+
+    - name: remove old data
+      file:
+        state: absent
+        path: "{{ item }}"
+      loop:
+        - .coverage
```

Comment:
again? why?

Reply:
The file is removed to ensure that we get fresh coverage data.

Reply:
yes, but why twice? (see line 31). Does any operation in-between recreate it?

Reply:
Good catch, it was wrong. Also it should have used the variable. It is all fixed and squashed now.